designs/static-capacity.md

```diff
@@ -0,0 +1,202 @@
+```yaml
+- key: topology.kubernetes.io/zone
+  operator: In
+  values: ["us-west-2a", "us-west-2b", "us-west-2c"]
+```
+
+This ensures the nodes are evenly distributed unless the user explicitly narrows the zone selection to a single AZ.
+
+## Example 
+
+```yaml
+apiVersion: karpenter.sh/v1beta1
+kind: NodePool
+metadata:
+  name: static-prod-nodepool
+spec:
+  replicas: 12 # Maintains 12 nodes regardless of pod demand
+  template:
+    metadata:
+      labels:
+        nodepool-type: static
+    spec:
+      nodeClassRef:
+        group: karpenter.k8s.aws
+        kind: EC2NodeClass
+        name: myEc2NodeClass
+      expireAfter: 720h
+      taints:
+        - key: example.com/special-taint
+          effect: NoSchedule
+      requirements:
+        - key: topology.kubernetes.io/zone
+          operator: In
+          values: ["us-west-2a", "us-west-2b", "us-west-2c"]
+        - key: karpenter.k8s.aws/instance-type
+          operator: In
+          values: ["m5.2xlarge"]
+  disruption:
+    budgets:
+      - nodes: 10%
+      # On Weekdays during business hours, don't do any deprovisioning.
+      - schedule: "0 9 * * mon-fri"
+        duration: 8h
+        nodes: "0"
+  limits:
+    # Constrain the size of static Nodepool 
+    nodes: 20 # Karpenter will not scale up nodes or create replacements if limits are breaching
+```
+
+We will support scaling static node pools using a command such as:
+
+```sh
+kubectl scale nodepool <name> --replicas=10
+```
+
+## Implementation Details
+
+### Controller Architecture
+
+We will be creating Static Capacity by adding 2 new controllers under a feature flag. This will be added to v1alpha1 release to get feedback from users.
+
+#### Scale-Up Controller
+- **Purpose**: Scale up nodeclaims to meet desired replicas
+- **Behavior**: Monitors the difference between desired replicas and current healthy NodeClaims, provisioning new nodes as needed respecting limits.nodes value
+- **Instance Selection**: Will find the cheapest instance possible during scale up operations based on the requirements
+- **Consolidation**: Static NodePools will not be consolidated to maintain predictable capacity
+
+#### Deprovisioning Controller  
+- **Purpose**: Scale down replicas when desired count is reduced
+- **Node Selection**: Will prioritize empty nodes for termination. In future iterations, we will add intelligent algorithms for selecting which nodes to scale down
+- **Termination Behavior**: Will forcefully terminate nodes to meet desired size without respecting PodDisruptionBudgets or terminationGracePeriod
+
+**Important Note**: The scale-down of replicas represents scaling down capacity, not disruption. We don't want to end up in a situation where a user sets the replicas to a lower number and we would still be running significantly more capacity due to blocking pods, PDBs, and terminationGracePeriod constraints.
```

Comment:
I disagree with this - by updating replicas you are setting a desired state. Karpenter should use it's existing termination mechanisms to get into that desired state. Customers have levers to pull if they want to ensure quick disruption, if they need to pull those levers they can. Making the behavior diverge between static and dynamic nodepools is surprising to me IMO.

Reply:
> We can in-future give a way to force scale-down so that we dont assume force kill intent by default.

FWIW this exists today - you can set an annotation on the NodeClaim to set a forced termination time. This isn't documented since it's an implementation details of TerminationGracePeriod, but it wouldn't take much to flesh it out and make it user facing. This is what NodeRepair already uses today.